main/parsers/zettle_parser.py

```diff
@@ -1,71 +1,89 @@
 # Use to generate the JSON-object from the Zettle API respnose.
-import sys, os
-
-from datetime import datetime, timedelta, time
-import pytz
-from dateutil.tz import tzlocal
-
-from requests_oauthlib import OAuth2Session
 import json
+import os
+import time
+from datetime import datetime, timedelta
+
 import dateutil.parser
+import requests
+from dateutil.tz import tzlocal
 from flatten_dict import unflatten
-
-from parsers.parser import Parser
 from helpers.asset_loader import AssetLoader
+from parsers.parser import Parser
 
 al = AssetLoader()
 
 
 class ZettleParser:  # Parser
     def __init__(self):
-        self.zettle = self.set_cred()
+        self.auth_token = None
 
     def intakt_type(self):
         return "Zettle"
 
-    def set_cred(self):
-        access_file = (
+    def get_new_auth_token(self):
+        access_file_path = (
             os.path.dirname(os.path.realpath(__file__)) + "/../credentials/access.json"
         )
+        with open(access_file_path) as access_file:
+            access_json = json.load(access_file)
```

Comment:
`access_file` should be closed here.

Reply:
The `with` statement closes the file automatically, that's the point of using `with` when opening files.